### DISCUSSION TITLE
Prevent X Axis Timestamps from Overlapping in Plots

### DISCUSSION THREAD

**Discussion by keenanjohnson** [zauberzeug/nicegui#3237] [Q&A]
### Question

Hey all! 

I noticed in the docs for [line plots](https://nicegui.io/documentation/section_data_elements#line_plot), that the timestamps overlap when you turn on the live plots and use a timestamp for the X-Axis. See image below from the docs page.

![image](https://github.com/zauberzeug/nicegui/assets/2559382/b9127283-7a64-4a95-bc8d-f0f83b58edfc)

I've reproduced this in my own project and I am wondering how to fix it? Thanks!

**Comment by falkoschindler** [comment]
Hi @keenanjohnson,

I guess this is more or less a Matplotlib question and you'll find plenty of examples online on how to fine-tune a Matplotlib figure. Here's an example changing the figure size, rotating the labels and increasing the bottom margin:

```py
line_plot = ui.line_plot(n=2, limit=20, figsize=(10, 5), update_every=5) \
    .with_legend(['sin', 'cos'], loc='upper center', ncol=2)

line_plot.fig.gca().tick_params(axis='x', rotation=-30)
line_plot.fig.subplots_adjust(bottom=0.25)
line_plot.update()
```
<img width="808" alt="Screenshot 2024-06-21 at 15 50 37" src="https://github.com/zauberzeug/nicegui/assets/5767091/553b6801-1444-4286-ae45-edb48b52eda5">

**Comment by keenanjohnson** [comment]
FYI I figured out this issue for anyone in the future. It seems that this happens if the update_every argument is set to 1 and you try to update the plot with an array of only one value. This script for example will throw the warning:

```py
import math
from datetime import datetime
from nicegui import ui

line_plot = ui.line_plot(n=2, limit=20, figsize=(3, 2), update_every=1) \
    .with_legend(['sin', 'cos'], loc='upper center', ncol=2)

def update_line_plot() -> None:
    now = datetime.now()
    x = now.timestamp()
    y1 = math.sin(x)
    y2 = math.cos(x)
    line_plot.push([now], [[y1], [y2]])

line_updates = ui.timer(0.1, update_line_plot, active=False)
line_checkbox = ui.checkbox('active').bind_value(line_updates, 'active')

ui.run()
```
